sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/DefaultAzureMonitorOptions.cs

```diff
@@ -24,19 +24,25 @@ public DefaultAzureMonitorOptions(IConfiguration? configuration = null)
 
         public void Configure(AzureMonitorOptions options)
         {
-            if (_configuration != null)
-            {
-                _configuration.GetSection(AzureMonitorSectionFromConfig).Bind(options);
-            }
-
             try
             {
-                // TODO: should we replace this with IConfiguration?
-                string connectionString = Environment.GetEnvironmentVariable(ConnectionStringEnvironmentVariable);
-
-                if (!string.IsNullOrWhiteSpace(connectionString))
+                if (_configuration != null)
                 {
-                    options.ConnectionString = connectionString;
+                    _configuration.GetSection(AzureMonitorSectionFromConfig).Bind(options);
+
+                    // IConfiguration can read from EnvironmentVariables or InMemoryCollection if configured to do so.
+                    var connectionString = _configuration[ConnectionStringEnvironmentVariable];
+                    if (!string.IsNullOrEmpty(connectionString))
+                    {
+                        options.ConnectionString = connectionString;
```

Comment:
please make sure this has a test case covering the behavior.

Reply:
Already covered! :)

https://github.com/Azure/azure-sdk-for-net/blob/d4cb1a2ed169944117b5d810a686ae6470d100f3/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/DefaultAzureMonitorOptionsTests.cs#L110-L132